src/main/java/hudson/plugins/sauce_ondemand/SauceEnvironmentUtil.java

```diff
@@ -209,4 +212,10 @@ public static String getBuildName(AbstractBuild<?, ?> build) {
         }
         return builder.toString();
     }
+
+    public static String generateTunnelIdentifier(final AbstractBuild build) {
+        String rawName = build.getProject().getName();
+        String sanitizedName = rawName.replaceAll(PATTERN_DISALLOWED_TUNNEL_ID_CHARS, "_");
```

Comment:
another thing - why not just use 
    
    UUID.randomUUID()

?

this will generate a string that we can use without worrying about collisions. also, we don't need any tests for it at all.